ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto

```diff
@@ -80,16 +81,19 @@ message SubmitAndWaitForTransactionIdResponse {
   // Must be a valid LedgerString (as described in ``value.proto``).
   // Required
   string transaction_id = 1;
+  google.protobuf.StringValue completion_offset = 2;
```

Comment:
This is mainly so that the offset is set as optional in the types. 

In most cases, it should be set actually (I think in all cases), but we're forced to use it as optional because of the types being used in the internal code for the completion.

Reply:
What do you mean by this part?
> we're forced to use it as optional because of the types being used in the internal code for the completion.

Reply:
Internally our type for the offset is `Option`, therefore we have to expose it like that through the API as well.

Reply:
I don't see a reason to introduce StringValue to signal a missing completion_offset. An empty string is perfectly conveying that, because an empty string is not a valid offset. Please see also
https://github.com/digital-asset/daml/blob/c9692666ae1dd60aa44e18bd48f92282d9b8c6e7/ledger-api/grpc-definitions/com/daml/ledger/api/v1/active_contracts_service.proto#L54
Also, all fields in the proto files must be documented in comments and indicated whether they are optional or required

Reply:
@mziolekda I changed the type to `string` so that we are consistent in our interfaces.

I do still believe that the right choice would be StringValue, having no value and having an invalid value are 2 totally different cases that should be represented by our type system.